app/serializers/api/admin/subscription_customer_serializer.rb

```diff
@@ -0,0 +1,20 @@
+module Api
+  module Admin
+    # Used by admin subscription form
+    # Searches for a ship and bill addresses for the customer
+    # where they are not already explicitly set
+    class SubscriptionCustomerSerializer < CustomerSerializer
+      def bill_address
+        finder.bill_address
+      end
+
+      def ship_address
+        finder.ship_address
+      end
+
+      def finder
+        @finder ||= OpenFoodNetwork::AddressFinder.new(object, object.email)
+      end
+    end
+  end
+end
```

Comment:
Is this related to the discussion you opened in the community? (I totally forgot about it)

Reply:
Yep, I'm still not really sure what to do about it, but this is the current convention, so I just stuck with it. I would quite like to decide whether we are going to stick with this approach or whether we are going to take a different path...

Reply:
Let's take a decision in the community thread then